conda/core/index.py

```diff
@@ -182,7 +182,7 @@ def _supplement_index_with_system(index):
 def get_archspec_name():
     from conda.base.context import _arch_names, _platform_map, non_x86_machines
 
-    target_plat, target_arch = context.subdir.split("-")
+    target_plat, target_arch = context.subdir.split("-", maxsplit=1)
```

Comment:
With a platform of "linux-64" this correctly parses the arch into `64` and in line 192 returns "x86_64".

With this change applied, it may fix the `ValueError` reported in the ticket, but with the custom platform of `linux-64-cxx03` won't be returning `x86_64` anymore but `None`. @chenghlee do you have a suggestion how to handle custom platforms like this?

Reply:
> do you have a suggestion how to handle custom platforms like this?

I would recommend custom platforms replace `-`s with `_`s in the [OS] platform and [hardware] architecture strings. Adding `maxsplit=1` smells like the wrong thing to do, because:

1. It implicitly asserts that _only_ the architecture component can have a `-`, which [I suspect] will inevitably lead to "but why can't my OS/platform have a `-` instead/as well?!" somewhere down the road.
2. "`subdir` contains one and only one `-`" has been around so long that it's effectively an undocumented but fundamental assumption throughout the conda ecosystem. So while adding `maxsplit` would solve  the user's _immediate_ problem, allowing `-` in either the platform or architecture subdir component will likely break something else in the conda ecosystem.

TL;DR: Supporting `-` in either the platform or architecture component of `subdir` is an ecosystem-wide, CEP-level initiative, and _not_ something we should add in a single line of code.

Reply:
I agree with Cheng here. We definitely also encode this assumption in mamba & rattler. Using another character would be better IMO.

Reply:
Since I'm not totally sure what detecting the platform is actually used for, is the suggestion to use `linux-64_cxx03` instead?  Wouldn't that still fail to detect as `x86_64`?  I'm also curious, what's the harm of failing to detect the platform correctly?